### PR TITLE
MangoQuery inspired queries

### PR DESCRIPTION
**Pull request by ivanslabbert**
Implement MangoQuery inspired queries in APP (1'st stage)

---

> [!NOTE]
> **Medium Risk**
> Moderate risk because it changes server-side query validation/rewriting and client-side IndexedDB query paths, which can affect access control filtering and content visibility. The changes are covered by new unit tests but could still impact edge-case selector shapes and index usage.
> 
> **Overview**
> **API querying now normalizes selectors to explicit `$and` form** via new `expandMangoSelector`, enabling top-level `$or` and making permission injection consistent; `QueryService` now extracts/validates `type`/`parentType`/`docType` from `$and`, enforces simple-equality values, and always appends the permission `memberOf` filter as an `$and` condition.
> 
> **App data fetching is migrated from Dexie `.where().filter()` to Mango queries** using `mangoToDexie`, with a new `mangoIsPublished` helper (and tests) to centralize publish/expiry/language-priority logic; this touches Explore/Home/Video pages, Bookmarks, related/tag viewers, and policy/copyright fetches. Startup now calls `warmMangoCaches()` and updates IndexedDB `docsIndex` definitions to better support the new query shapes, alongside new component-level tests for pinned/unpinned content and continue-watching ordering.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1f3e70996ed4f9637151438db2af7bfa333ab66f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: shared/src/util/MangoQuery/mangoCompile.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ */
+export function mangoCompile(q: MangoSelector): Predicate {
+    // Handle invalid query (not cacheable)
+    if (q === null || typeof q !== "object") {
+        return () => false;
+    }
+
+    // Handle empty query (not cacheable, just return constant)
+    let hasKeys = false;
+    for (const _ in q) {
```
Comment (MrDirkelz) [inline shared/src/util/MangoQuery/mangoCompile.ts]: Same here for "_"
